src/VintageBookshelf.Api/Controllers/BooksController.cs

```diff
@@ -49,17 +49,57 @@ public async Task<ActionResult<IEnumerable<BookDto>>> Add([FromBody] BookDto boo
             return CustomResponse(bookDto);
         }
         
+        [HttpPut("{id:long}")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(400)]
+        public async Task<ActionResult<BookDto>> Update(long id, [FromBody] BookDto bookDto)
+        {
```

Comment:
I believe you don't need to use [FromBody] in this case. The object would be used for that even without the attribute.

Reply:
This is mostly for displaying on Swagger that a body payload is required for this endpoint as an input. If you don't use `FromBody` attribute, then it won't be displayed on Swagger. Perhaps there's another option for this instead?

Reply:
Not sure. I think it is worth investigating once the framework has its pre conditions to associate the request with the parameters in the action.

Reply:
It's not an issue at the code level, I agree. It's mostly for documentation purposes and to show the correct body input content type (`application/json`).